lib/inspec/resources/ssh_key.rb

```diff
@@ -0,0 +1,129 @@
+    supports platform: "windows"
+
+    desc "public/private SSH key pair test"
+
+    example <<~EXAMPLE
+      describe ssh_key('path: ~/.ssh/id_rsa') do
+        its('key_length') { should eq 4096 }
+        its('type') { should cmp /rsa/ }
+        it { should be_private }
+      end
+    EXAMPLE
+
+    include FileReader
+
+    def initialize(keypath, passphrase = nil)
+      skip_resource "The `ssh_key` resource is not yet available on your OS." unless inspec.os.unix? || inspec.os.windows?
+      @key_path = set_ssh_key_path(keypath)
+      @passphrase = passphrase
+      @key = read_ssh_key
+      super(@key_path)
+    end
+
+    def public?
+      return if @key.nil?
+
+      @key[:public]
+    end
+
+    def private?
+      return if @key.nil?
+
+      @key[:private]
+    end
+
+    def key_length
+      return if @key.nil?
+
+      @key[:key_length]
+    end
+
+    def type
+      return if @key.nil?
+
+      @key[:type]
+    end
+
+    # Define a resource ID. This is used in reporting engines to uniquely identify the individual resource.
+    # This might be a file path, or a process ID, or a cloud instance ID. Only meaningful to the implementation.
+    # Must be a string. Defaults to the empty string if not implemented.
+    def resource_id
+      @key_path || "SSH key"
+    end
+
+    def to_s
+      "ssh_key #{@key_path}"
+    end
+
+    private
+
+    def set_ssh_key_path(keypath)
+      if File.exist?(keypath)
+        @key_path = keypath
+      elsif File.exist?(File.join("#{Dir.home}/.ssh/", keypath))
+        @key_path = File.join("#{Dir.home}/.ssh/", keypath)
+      else
+        raise Inspec::Exceptions::ResourceSkipped, "Can't find file: #{keypath}"
+      end
+    end
+
+    def read_ssh_key
+      key_data = {}
+      key = nil
+      filecontent = read_file_content((@key_path), @passphrase)
+      raise Inspec::Exceptions::ResourceSkipped, "File is empty: #{@key_path}" if filecontent.split("\n").empty?
+
+      if filecontent.split("\n")[0].include?("PRIVATE")
+        # Net::SSH::KeyFactory does not have support to load private key for DSA
+        key = Net::SSH::KeyFactory.load_private_key(@key_path, @passphrase, false)
+        key_data[:private] = true unless key.nil?
+        key_data[:public] = false unless key.nil?
+      else
+        key = Net::SSH::KeyFactory.load_public_key(@key_path)
+        key_data[:public] = true unless key.nil?
+        key_data[:public] = true unless key.nil?
+      end
+
+      unless key.nil?
+        # The data send for ssh type is not in same format so it's good to match on the string
+        key_data[:type] = key.ssh_type
+        key_data[:key_length] = key_lengh(key)
+      end
+
+      key_data
+    rescue OpenSSL::PKey::PKeyError => e
+      raise Inspec::Exceptions::ResourceFailed, "#{e.message}"
+    end
+
+    def key_lengh(key)
+      case key.class.to_s
+      when "OpenSSL::PKey::RSA"
+        key.public_key.n.num_bits
+      when "OpenSSL::PKey::EC"
```

Comment:
Since for all other key types, we are returning value `nil`, then instead of having these case checks, can we not pass nil by having one default `else` condition?
So something like
```
case key.class.to_s
when "OpenSSL::PKey::RSA"
     key.public_key.n.num_bits
else
    nil
end
```
Will this cause any issue?